File: BT Tool/markform old/markform/frm_sub_report_log.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

using Newtonsoft.Json;

namespace markform
{
    public partial class frm_sub_report_log : Form
    {
        CustomClass cf = new CustomClass();
        public frm_sub_report_log()
        {
            InitializeComponent();
            this.Tag = "breaklog";
        }
        public void SetLocation(int mainfrmwidth)
        {
            this.Location = new Point((Screen.PrimaryScreen.WorkingArea.Width - (this.Width + mainfrmwidth)), (Screen.PrimaryScreen.WorkingArea.Height - this.Height));
        }
        public void DisplayData(string breakdata, string username)
        {
            this.Text = "Break Log - " + username.ToUpper();
            List<BreakLogData> bdata = JsonConvert.DeserializeObject<List<BreakLogData>>(breakdata);
            DataTable dt = new DataTable();
            dt.Columns.Add("Filename", typeof(string));
            dt.Columns.Add("Break Start", typeof(string));
            dt.Columns.Add("Break End", typeof(string));
            dt.Columns.Add("Duration", typeof(string));
            double dur = 0;
            foreach(BreakLogData bld in bdata)
            {
                dt.Rows.Add(bld.filename, bld.bstart, bld.bend, bld.duration);
                dur = dur + TimeSpan.Parse(bld.duration).TotalSeconds;
            }
            dgv_breaklog.DataSource = dt;
            tss_total_dur.Text = "Total Duration: " + cf.SecondsToTime24(dur);

            tss_total_dur.ForeColor = (dur > 5400)?Color.Red:Color.Black;
            cf.RowsNumber(dgv_breaklog);
        }
        private void dgv_breaklog_SelectionChanged(object sender, EventArgs e)
        {
            GetSelectedTotalDur();
        }

        private void GetSelectedTotalDur()
        {
            double dur = 0;
            foreach (DataGridViewRow row in dgv_breaklog.SelectedRows)
            {
                dur = dur + TimeSpan.Parse(row.Cells[3].Value.ToString()).TotalSeconds;
            }
            tss_sel_dur.Text = "Selected Total Duration: " + cf.SecondsToTime24(dur);
        }
    }
}

File: BT Tool/BT Tool/Forms/frm_sub_report_log.vb
Imports System
Imports System.Collections.Generic
Imports System.Data
Imports System.Drawing
Imports System.Windows.Forms
Imports Newtonsoft.Json

Public Class frm_sub_report_log


#Region "Class"
    Private cf As New markform.CustomClass
#End Region

#Region "Variables"
    Public Sub New()

        ' This call is required by the designer.
        InitializeComponent()

        ' Add any initialization after the InitializeComponent() call.
        Me.Tag = "breaklog"
    End Sub
#End Region

#Region "sub"

    Public Sub SetLocation(ByVal mainfrmwidth As Integer)
        Me.Location = New Point((Screen.PrimaryScreen.WorkingArea.Width - (Me.Width + mainfrmwidth)), (Screen.PrimaryScreen.WorkingArea.Height - Me.Height))
    End Sub

    Private Sub dgv_breaklog_SelectionChanged(sender As Object, e As EventArgs) Handles dgv_breaklog.SelectionChanged
        GetSelectedTotalDur()
    End Sub

    Public Sub DisplayData(ByVal breakdata As String, ByVal username As String)
        Me.Text = "Break Log - " & username.ToUpper
        ' Dim bdata As List(Of BreakLogData) = JsonConvert.DeserializeObject(Of BreakLogData)(breakdata)
        Dim bdata As List(Of BreakLogData) = JsonConvert.DeserializeObject(Of List(Of BreakLogData))(breakdata)
        Dim dt As New DataTable
        dt.Columns.Add("Filename", GetType(String))
        dt.Columns.Add("Break Start", GetType(String))
        dt.Columns.Add("Break End", GetType(String))
        dt.Columns.Add("Duration", GetType(String))
        Dim dur As Double = 0
        For Each bld As BreakLogData In bdata
            dt.Rows.Add(bld.filename, bld.bstart, bld.bend, bld.duration)
            dur = dur & TimeSpan.Parse(bld.duration).TotalSeconds
        Next
        dgv_breaklog.DataSource = dt
        tss_total_dur.Text = "Total Duration: " & cf.SecondsToTime24(dur)
        tss_total_dur.ForeColor = If(dur > 5400, Color.Red, Color.Black)
        cf.RowsNumber(dgv_breaklog)
    End Sub

    Public Sub GetSelectedTotalDur()
        Dim dur As Double = 0
        For Each row As DataGridViewRow In dgv_breaklog.SelectedRows
            dur = dur & TimeSpan.Parse(row.Cells(3).Value.ToString()).TotalSeconds
        Next
        tss_sel_dur.Text = "Selected Total Duration: " & cf.SecondsToTime24(dur)
    End Sub
#End Region
End Class